app/core/model/artifact.py

```diff
@@ -0,0 +1,214 @@
+import base64
+from dataclasses import dataclass, field
+from enum import Enum
+import json
+from typing import Any, Dict, Optional
+
+
+class ArtifactStatus(Enum):
+    """Enum representing the possible states of an artifact."""
+
+    CREATED = "CREATED"
+    PROCESSING = "PROCESSING"
+    FINISHED = "FINISHED"
+    FAILED = "FAILED"
+    ARCHIVED = "ARCHIVED"
+    STOPPED = "STOPPED"
+
+
+class ContentType(Enum):
```

Comment:
It is not recommended to adopt the HTML design. Currently, `Graph` is enough. Just use ordinary enumeration.